src/Workspaces/Core/Portable/Workspace/Solution/SolutionInfo.cs

```diff
@@ -160,7 +156,16 @@ public static SolutionAttributes ReadFrom(ObjectReader reader)
             }
 
             public Checksum Checksum
-                => _lazyChecksum ??= Checksum.Create(this, static (@this, writer) => @this.WriteTo(writer));
+            {
+                get
+                {
+                    // Ensure reading/writing from the nullable checksum is atomic.
```

Comment:
Small request: add a comment to the _lazyChecksum fields too saying to take a lock on 'this'; we don't usually do that (in favor of an explicit gate or something) so the pattern is somewhat less common. That said I have no concerns with this as written.